components/build/tekton-chains/tekton-chains-controller-shared-secret.yaml

```diff
@@ -0,0 +1,28 @@
+---
```

Comment:
This should only be applied if the SharedSecret named `redhat-appstudio-user-workload` actually exists. My understanding is that this only happens in the staging environment.

This PR is in draft because I need to either:
1. Figure out how to apply an overlay to the tekton-chains Deployment only in the staging environments; or
2. Always create the `redhat-appstudio-user-workload` SharedSecret for preview and development modes.

Reply:
Ok, it looks like `SharedSecret` is already created in components/build/shared-resources/redhat-appstudio-user-workload-sharedsecret.yaml

So let's take approach 2 and this PR is good as is. Moving it out of draft.

Reply:
It seems like if the `SharedSecret` is used when the underlying `Secret` does not exist, the mounted path is always a directory. This causes the `k8schain` library used by Chains to error - meaning no image signature nor attestation. 

I submitted a [fix](https://github.com/google/go-containerregistry/pull/1420 ) for the library, but I have no idea how long that will take to get merged and released. I can create a fork but would really prefer not to do that...

@Michkov, can we always create the `build-templates/redhat-appstudio-user-workload` secret? I think an "empty" one should be enough.

Reply:
I think I have a workaround. Waiting for e2e test results.